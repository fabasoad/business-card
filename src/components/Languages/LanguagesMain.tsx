import * as React from 'react'
import { useTranslation } from 'react-i18next'
import LanguageLinkItem from './LanguageLinkItem'
import LanguageTextItem from './LanguageTextItem'

export default function LanguagesMain() {
  const { t } = useTranslation()
  return (
    <div id="languages" className="light-component">
      <div className="section-title text-center center">
        <h2>{t('business-card-languages-title')}</h2>
        <hr />
      </div>
      <div className="container">
        <div className="row">
          <div className="language-list col text-center">
            <LanguageLinkItem code="jp" text="JLPT N5" url="https://drive.google.com/file/d/1wDeulFdf9-8DBa_k0F8tiRUtdlG7mh3L/view?usp=sharing" />
            <LanguageLinkItem code="gb" text="TOEIC 825" url="https://drive.google.com/file/d/1nzM9WQqR7PQpHKDiWvkRYtcNbnRW8-OB/view?usp=sharing" />
            <LanguageTextItem code="ru" text={t('business-card-languages-native')} />
            <LanguageTextItem code="ua" text={t('business-card-languages-native')} />
          </div>
        </div>
      </div>
    </div>
  )
}
