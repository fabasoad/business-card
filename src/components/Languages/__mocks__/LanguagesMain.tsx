import * as React from 'react'

export default function LanguagesMain() {
  return (
    <div data-testid="LanguagesMain"></div>
  )
}
