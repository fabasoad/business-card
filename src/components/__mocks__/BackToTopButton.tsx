import * as React from 'react'

export default function BackToTopButton() {
  return (
    <div data-testid="BackToTopButton"></div>
  )
}
