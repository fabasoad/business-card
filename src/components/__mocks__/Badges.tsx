import * as React from 'react'

export default function Badges() {
  return (
    <div data-testid="Badges"></div>
  )
}
