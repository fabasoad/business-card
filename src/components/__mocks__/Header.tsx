import * as React from 'react'

export default function Header() {
  return (
    <div data-testid="Header"></div>
  )
}
