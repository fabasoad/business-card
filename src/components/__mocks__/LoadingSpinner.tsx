import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
