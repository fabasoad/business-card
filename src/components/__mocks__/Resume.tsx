import * as React from 'react'

export default function Resume() {
  return (
    <div data-testid="Resume"></div>
  )
}
