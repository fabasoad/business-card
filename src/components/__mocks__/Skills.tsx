import * as React from 'react'

export default function Skills() {
  return (
    <div data-testid="Skills"></div>
  )
}
