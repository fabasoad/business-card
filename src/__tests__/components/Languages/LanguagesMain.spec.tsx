/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import LanguagesMain from '../../../components/Languages/LanguagesMain'
import { useTranslation } from '../../__mocks__/react-i18next'

let tMock

beforeAll(() => {
  tMock = useTranslation().t
})

test('should render LanguageItem correctly', () => {
  const wrapper: ShallowWrapper = shallow(<LanguagesMain />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toHaveBeenCalledTimes(3)
  expect(tMock).toHaveBeenNthCalledWith(1, 'business-card-languages-title')
  new Array<number>(2, 3).forEach(
    (n) => expect(tMock).toHaveBeenNthCalledWith(
      n, 'business-card-languages-native'
    )
  )
})
