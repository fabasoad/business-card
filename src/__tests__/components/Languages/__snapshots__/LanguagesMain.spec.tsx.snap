// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LanguageItem correctly 1`] = `
<div
  className="light-component"
  id="languages"
>
  <div
    className="section-title text-center center"
  >
    <h2>
      business-card-languages-title
    </h2>
    <hr />
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="language-list col text-center"
      >
        <LaguageLinkItem
          code="jp"
          text="JLPT N5"
          url="https://drive.google.com/file/d/1wDeulFdf9-8DBa_k0F8tiRUtdlG7mh3L/view?usp=sharing"
        />
        <LaguageLinkItem
          code="gb"
          text="TOEIC 825"
          url="https://drive.google.com/file/d/1nzM9WQqR7PQpHKDiWvkRYtcNbnRW8-OB/view?usp=sharing"
        />
        <LanguageTextItem
          code="ru"
          text="business-card-languages-native"
        />
        <LanguageTextItem
          code="ua"
          text="business-card-languages-native"
        />
      </div>
    </div>
  </div>
</div>
`;
